src/System.Security.Principal.Windows/tests/SecurityIdentifierTests.cs

```diff
@@ -66,21 +73,13 @@ public void ValidateToStringUsingWhoami()
 
         if (whoamiProcess.ExitCode == 0)
         {
-            int startSid = output.IndexOf("S-1-5-");
-            int endSid = 0;
-            if (startSid >= 0)
-            {
-                int length = Math.Min(output.Length - startSid, librarySid.Length);
-                windowsSid = output.Substring(startSid, length);
-
-                if (output.Length > startSid + length)
-                {
-                    Assert.True(char.IsWhiteSpace(output[startSid + length]));
-                }
-            }
-            if (endSid > startSid)
+            int startSid = output.IndexOf("S-1-");
+            Assert.InRange(startSid, 0, int.MaxValue);
+            int length = Math.Min(output.Length - startSid, librarySid.Length);
+            windowsSid = output.Substring(startSid, length);
+            if (output.Length > startSid + length)
             {
-                windowsSid = output.Substring(startSid, (endSid - startSid));
+                Assert.True(char.IsWhiteSpace(output[startSid + length]));
             }
         }
         Assert.NotNull(windowsSid);
```

Comment:
These asserts are going to fail if the whoamiProcess didn't succeed, right?  So rather than the if block above, shouldn't it just be asserting that `Assert.Equal(0, whoamiProcess.ExitCode);`?

Reply:
GitHub won't let me comment on it, but wouldn't it be better for the `Assert.NotNull(librarySid)` and `Assert.NotEmpty(librarySid)` to be moved to the top of the method where `librarySid` is initialized?  You can then also remove the `Assert.NotEmpty(windowsSid)`, as that'll be validated as part of the `Assert.Equal(windowsSid, librarySid)`.